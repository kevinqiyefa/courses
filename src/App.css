.App {
  text-align: center;
}

.disabled:hover {
  cursor: not-allowed;
}
