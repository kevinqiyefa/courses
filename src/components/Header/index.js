import React from 'react';
import { NavLink } from 'react-router-dom';

export default function Header() {
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
      <div className="container pl-0 pr-0 pt-2 pb-2">
        <NavLink className="navbar-brand" to="/">
          LOGO
        </NavLink>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarColor01"
          aria-controls="navbarColor01"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"></span>
        </button>

        <div className="collapse navbar-collapse" id="navbarColor01">
          <ul className="navbar-nav mr-auto">
            <li className="nav-item">
              <NavLink className="nav-link" to="/">
                Courses
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
}
