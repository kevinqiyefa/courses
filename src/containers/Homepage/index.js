import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import {
  fetchCoursesRequest,
  addCourseRequest
} from '../../store/actions/courses';
import { fetchStudentsRequest } from '../../store/actions/students';
import Loader from '../../components/Loader';
import CourseCards from '../../components/CourseCards';
import AddCourse from '../../components/AddCourse';
import CoursesWrapper from '../../components/CoursesWrapper';

function Homepage() {
  const { courses, loading } = useSelector(state => state.course);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!courses.length) {
      dispatch(fetchCoursesRequest());
      dispatch(fetchStudentsRequest());
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const sortByIdCourses = courses.sort((a, b) => b.id - a.id);
  const content = loading ? (
    <Loader />
  ) : (
    <CourseCards courses={sortByIdCourses} />
  );
  const title = 'Hello, Professor!';
  const description =
    "Welcome back! Here're your classes and students for the school year.";

  const addCourse = state => dispatch(addCourseRequest(state));

  return (
    <CoursesWrapper title={title} description={description}>
      <AddCourse addCourse={addCourse} />
      {content}
    </CoursesWrapper>
  );
}

export default Homepage;
