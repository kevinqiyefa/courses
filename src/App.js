import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';

import Notfound from './components/NotFoundPage';
import { configureStore } from './store';
import Homepage from './containers/Homepage';
import Header from './components/Header';
import Course from './containers/Course';
import Students from './containers/Students';

import './App.css';

const store = configureStore();

function App() {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Header />
        <div className="App">
          <Switch>
            <Route exact path="/" component={Homepage} />
            <Route exact path="/:courseID" component={Course} />
            <Route exact path="/:courseID/students" component={Students} />
            <Route component={Notfound} />
          </Switch>
        </div>
      </BrowserRouter>
    </Provider>
  );
}

export default App;
