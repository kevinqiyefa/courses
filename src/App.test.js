import React from 'react';
import App from './App';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './components/Header';
import Homepage from './containers/Homepage';

describe('<App />', () => {
  let wrapper;

  wrapper = shallow(<App />);

  it('shows a Header component', () => {
    expect(wrapper.find(Header).length).toEqual(1);
  });

  it('shows a Homepage component', () => {
    wrapper = mount(<App />);
    expect(wrapper.find(Homepage).length).toEqual(1);
  });

  it('matches snapshot', () => {
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
