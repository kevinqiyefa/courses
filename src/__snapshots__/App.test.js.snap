// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<App>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <div
          className="App"
        >
          <Header>
            <nav
              className="navbar navbar-expand-lg navbar-dark bg-primary"
            >
              <div
                className="container pl-0 pr-0 pt-2 pb-2"
              >
                <NavLink
                  className="navbar-brand font-weight-bold"
                  to="/"
                >
                  <Link
                    aria-current="page"
                    className="navbar-brand font-weight-bold active"
                    style={Object {}}
                    to="/"
                  >
                    <a
                      aria-current="page"
                      className="navbar-brand font-weight-bold active"
                      href="/"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      LOGO
                    </a>
                  </Link>
                </NavLink>
                <button
                  aria-controls="navbarColor01"
                  aria-expanded="false"
                  aria-label="Toggle navigation"
                  className="navbar-toggler"
                  data-target="#navbarColor01"
                  data-toggle="collapse"
                  type="button"
                >
                  <span
                    className="navbar-toggler-icon"
                  />
                </button>
                <div
                  className="collapse navbar-collapse"
                  id="navbarColor01"
                >
                  <ul
                    className="navbar-nav mr-auto"
                  >
                    <li
                      className="nav-item"
                    >
                      <NavLink
                        className="nav-link"
                        to="/"
                      >
                        <Link
                          aria-current="page"
                          className="nav-link active"
                          style={Object {}}
                          to="/"
                        >
                          <a
                            aria-current="page"
                            className="nav-link active"
                            href="/"
                            onClick={[Function]}
                            style={Object {}}
                          >
                            Courses
                          </a>
                        </Link>
                      </NavLink>
                    </li>
                  </ul>
                </div>
              </div>
            </nav>
          </Header>
          <Switch>
            <Route
              component={[Function]}
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              exact={true}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              path="/"
            >
              <Homepage
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <CoursesWrapper
                  description="Welcome back! Here're your classes and students for the school year."
                  title="Hello, Professor!"
                >
                  <div
                    className="jumbotron container mt-5"
                  >
                    <h1
                      className="display-3"
                    >
                      Hello, Professor!
                    </h1>
                    <p
                      className="lead mb-5"
                    >
                      Welcome back! Here're your classes and students for the school year.
                    </p>
                    <AddCourse
                      addCourse={[Function]}
                    >
                      <div
                        className="d-flex justify-content-end mb-5"
                      >
                        <button
                          className="btn btn-primary add-course-btn"
                          data-target="#addEditCourseModal"
                          data-toggle="modal"
                          type="button"
                        >
                          Add Course
                        </button>
                      </div>
                      <AddEditCourse
                        handleInputChange={[Function]}
                        handleSliderChange={[Function]}
                        handleSubmit={[Function]}
                        modalTitle="Adding a New Course!"
                        state={
                          Object {
                            "course_description": "",
                            "course_name": "",
                            "length": 45,
                            "studentIDs": Array [],
                            "subject": "English",
                          }
                        }
                        submitBtnText="Add Course"
                      >
                        <div
                          aria-hidden="true"
                          aria-labelledby="addEditCourseModalTitle"
                          className="modal fade"
                          id="addEditCourseModal"
                          role="dialog"
                          tabIndex="-1"
                        >
                          <div
                            className="modal-dialog modal-dialog-centered"
                            role="document"
                          >
                            <div
                              className="modal-content"
                            >
                              <div
                                className="modal-header"
                              >
                                <h5
                                  className="modal-title"
                                  id="exampleModalLongTitle"
                                >
                                  Adding a New Course!
                                </h5>
                                <button
                                  aria-label="Close"
                                  className="close close-add-course-modal"
                                  data-dismiss="modal"
                                  type="button"
                                >
                                  <span
                                    aria-hidden="true"
                                  >
                                    Ã—
                                  </span>
                                </button>
                              </div>
                              <div
                                className="modal-body"
                              >
                                <CourseForm
                                  handleInputChange={[Function]}
                                  handleSliderChange={[Function]}
                                  state={
                                    Object {
                                      "course_description": "",
                                      "course_name": "",
                                      "length": 45,
                                      "studentIDs": Array [],
                                      "subject": "English",
                                    }
                                  }
                                >
                                  <form>
                                    <fieldset
                                      className="text-left"
                                    >
                                      <div
                                        className="form-group"
                                      >
                                        <label
                                          htmlFor="courseName"
                                        >
                                          Course Name
                                        </label>
                                        <input
                                          className="form-control"
                                          id="courseName"
                                          name="course_name"
                                          onChange={[Function]}
                                          placeholder="Enter course name"
                                          type="text"
                                          value=""
                                        />
                                      </div>
                                      <div
                                        className="form-group"
                                      >
                                        <label
                                          htmlFor="selectSubject"
                                        >
                                          Select a subject
                                        </label>
                                        <select
                                          className="form-control"
                                          id="selectSubject"
                                          name="subject"
                                          onChange={[Function]}
                                          value="English"
                                        >
                                          <option
                                            value="Science"
                                          >
                                            Science
                                          </option>
                                          <option
                                            value="Math"
                                          >
                                            Math
                                          </option>
                                          <option
                                            value="English"
                                          >
                                            English
                                          </option>
                                          <option
                                            value="Art"
                                          >
                                            Art
                                          </option>
                                          <option
                                            value="Foreign Language"
                                          >
                                            Foreign Language
                                          </option>
                                          <option
                                            value="Social studies"
                                          >
                                            Social Studies
                                          </option>
                                          <option
                                            value="History"
                                          >
                                            History
                                          </option>
                                          <option
                                            value="Music"
                                          >
                                            Music
                                          </option>
                                          <option
                                            value="Physical education"
                                          >
                                            Physical education
                                          </option>
                                        </select>
                                      </div>
                                      <div
                                        className="form-group"
                                      >
                                        <label
                                          htmlFor="course_description"
                                        >
                                          Course Description
                                        </label>
                                        <textarea
                                          className="form-control"
                                          id="course_description"
                                          name="course_description"
                                          onChange={[Function]}
                                          placeholder="Enter the description"
                                          value=""
                                        />
                                      </div>
                                      <div
                                        className="form-group"
                                      >
                                        <label
                                          htmlFor="course_length"
                                        >
                                          Course length (minutes)
                                        </label>
                                        <Slider
                                          handleLabel=""
                                          id="course_length"
                                          labels={Object {}}
                                          max={120}
                                          min={0}
                                          onChange={[Function]}
                                          orientation="horizontal"
                                          reverse={false}
                                          step={1}
                                          tooltip={true}
                                          value={45}
                                        >
                                          <div
                                            aria-orientation="horizontal"
                                            aria-valuemax={120}
                                            aria-valuemin={0}
                                            aria-valuenow={45}
                                            className="rangeslider rangeslider-horizontal"
                                            onMouseDown={[Function]}
                                            onMouseUp={[Function]}
                                            onTouchEnd={[Function]}
                                            onTouchStart={[Function]}
                                          >
                                            <div
                                              className="rangeslider__fill"
                                              style={
                                                Object {
                                                  "width": "0px",
                                                }
                                              }
                                            />
                                            <div
                                              className="rangeslider__handle"
                                              onKeyDown={[Function]}
                                              onMouseDown={[Function]}
                                              onTouchEnd={[Function]}
                                              onTouchMove={[Function]}
                                              style={
                                                Object {
                                                  "left": "0px",
                                                }
                                              }
                                              tabIndex={0}
                                            >
                                              <div
                                                className="rangeslider__handle-label"
                                              />
                                            </div>
                                            <ul
                                              className="rangeslider__labels"
                                            />
                                          </div>
                                        </Slider>
                                      </div>
                                    </fieldset>
                                  </form>
                                </CourseForm>
                              </div>
                              <div
                                className="modal-footer"
                              >
                                <button
                                  className="btn btn-secondary"
                                  data-dismiss="modal"
                                  type="button"
                                >
                                  Close
                                </button>
                                <button
                                  className="btn btn-primary disabled"
                                  data-dismiss=""
                                  onClick={[Function]}
                                  type="button"
                                >
                                  Add Course
                                </button>
                              </div>
                            </div>
                          </div>
                        </div>
                      </AddEditCourse>
                    </AddCourse>
                    <Loader>
                      <div
                        className="Loader"
                      />
                    </Loader>
                  </div>
                </CoursesWrapper>
              </Homepage>
            </Route>
          </Switch>
        </div>
      </Router>
    </BrowserRouter>
  </Provider>
</App>
`;
